frontend/lint/tests/no-literal-metabase-strings.unit.spec.js

```diff
@@ -0,0 +1,149 @@
+import { RuleTester } from "eslint";
+import noLiteralMetabaseString from "../eslint-rules/no-literal-metabase-strings";
+
+const ruleTester = new RuleTester({
+  parserOptions: {
+    ecmaVersion: 2015,
+    sourceType: "module",
+    ecmaFeatures: { jsx: true },
+  },
+});
+
+const VALID_CASES = [
+  {
+    // "Metabase in import sources"
+    code: `
+import OpenInMetabase from "../components/OpenInMetabase";`,
+  },
+  {
+    // "Metabase in reexports"
+    code: `
+export { MetabaseLinksToggleWidget } from "./MetabaseLinksToggleWidget";`,
+  },
+  {
+    // "No Metabase string",
+    code: `
+  const label = "some string"`,
+  },
+  {
+    // "Detect in literal strings",
+    code: `
+  import { getApplicationName } from 'metabase/selectors/whitelabel';
```

Comment:
Why should we consider this as valid?
It allows this:
```tsx
import { t } from "ttag";
import { useSelector } from "metabase/lib/redux";
import { getApplicationName } from "metabase/selectors/whitelabel";

export function SomeComponent() {
  const appName = useSelector(getApplicationName);
  return (
    <div>
      <h1>{"Hello Metabase - hardcoded"}</h1>
      <h1>Metabase</h1>
      <h1>{t`Metabase`}</h1>
      <Something label="Metabase" />
      <Something label={t`Metabase`} />
      <h2>{appName}</h2>
    </div>
  );
}

const Something = ({ label }: { label: string }) => {
  return <h2>{label}</h2>;
};
```
where we us the applicationName once but can also hardcode it in other places

Reply:
It's a compromise for the rule to not get too complicated. Surely we can try to track where an imported name is being used, but that would definitely require more time to implement.

This simplifies the implementation so that we trust if users import the selector they will use it correctly.

At least this should catch the majority of the problem where people just forget to import `getApplicationName` from the start. Actually that won't help when you're adding a new string in the same file that already has the selector imported though.

I didn't want to spend too much time on this so I think this is good enough.

Reply:
I'm still not understanding why we check if that import is present, isn't the name returned by that selector supposed to replace the hardcoded "Metabase" string?
ie, if before it was
```tsx
const Component = () => {

  return <h1>t`Welcome to Metabase`<h1>
}
 ```
 
 it should be fixed as
 ```tsx
import { useSelector } from "metabase/lib/redux";
import { getApplicationName } from "metabase/selectors/whitelabel";

const Component = () => {
   const applicationName = useSelector(getApplicationName); 

  return <h1>t`Welcome to ${applicationName}`<h1>
}
 ```
which removes "Metabase" from the source code.

Shouldn't we be able to remove the import check entirely?

Reply:
Fixed in https://github.com/metabase/metabase/pull/38553/commits/9219375409a6a48697498ec1b8db7f852fd9cd07. This pretty much moves all the previous tests that are supposed to fail from `VALID_CASES` to `INVALID_CASES`